### PR TITLE
Added base region layout

### LINKED ISSUE #12: Responsive navigation
- Top and left app navigation

### PR DESCRIPTION
**Pull request by lesleychard**
Resolves #12 

- Added responsive top and left navigation
- Implemented Material UI x Next JS config to fix hot reload issues
- Added advanced `color` helpers
- A start to some theme configs (mostly palettes)
- Added a VS Code snippet for creating react components

### CODE REVIEW

File: components/Logo.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        classNameProp,
+      )}
+      {...other}
+    >
+      {t('app-name')}
+    </Component>
+  );
+};
+
+Logo.defaultProps = {
```
Comment (lesleychard) [inline components/Logo.tsx:57]: I realize this needs to go - updating this PR shortly

**Review by whymarrh — CHANGES_REQUESTED** [review]
Two small things:

- Can we add `.vscode/` to the `.gitignore`?
- Can we move the `@types/` packages to dev deps?

**Comment by lesleychard** [comment]
> Two small things:
> 
> * Can we add `.vscode/` to the `.gitignore`?
> * Can we move the `@types/` packages to dev deps?

We can, on both accounts. I left the `.vscode` in there because I thought the snippet I added for react components would come in handy for future development 🤷‍♀️

**Comment by lesleychard** [comment]
@whymarrh Should be good now.

**Review by whymarrh — DISMISSED** [review]
LGTM! Re: the VS Code files, I believe we can force add individual files as needed, adding the directory to the ignore list should prevent us from inadvertently adding files.

**Comment by lesleychard** [comment]
@whymarrh You good with this one now? I'll add the snippet back in if I feel we need it.

**Review by whymarrh — APPROVED** [review]
LGTM!

**Pull request merged**